northstar/src/npk/manifest/console.rs

```diff
@@ -14,11 +14,17 @@ use strum_macros::{EnumCount, EnumIter};
 #[skip_serializing_none]
 #[derive(Clone, PartialEq, Eq, Debug, Default, Serialize, Deserialize, JsonSchema)]
 #[serde(deny_unknown_fields)]
-pub struct ConsoleConfiguration {
+pub struct Configuration {
     /// Permissions
     pub permissions: Permissions,
     /// Limits the number of requests processed per second
-    pub max_requests_per_sec: Option<u32>,
+    pub max_requests_per_sec: Option<usize>,
+    /// Maximum request size in characters
+    pub max_request_size: Option<usize>,
+    /// Maximum npk stream size in bytes
+    pub max_install_stream_size: Option<u64>,
+    /// NPK stream timeout in seconds
+    pub npk_stream_timeout: Option<u64>,
```

Comment:
just wondering, I know that this is easy to implement by adding a timeout to the `network_stream.next()` call, but wouldn't it make more sense to have a timeout in the overall npk transmission? with this method an npk could be transmitted in the worst case in time `#packages * timeout_duration`  and without knowing `#packages` is impossible to have some sense on how long would it take :man_shrugging:

Reply:
That's a fair point. The problem with the installs is already that, once active it blocks the runtimes main loop. This needs to be improved anyways and the timeout handling will be part of that.